core/rawdb/accessors_chain.go

```diff
@@ -542,23 +541,23 @@ func DeleteTermini(db ethdb.KeyValueWriter, hash common.Hash) {
 }
 
 // ReadPendingHeader retreive's the pending header stored in hash.
-func ReadPendingHeader(db ethdb.Reader, hash common.Hash) *types.Header {
+func ReadPendingHeader(db ethdb.Reader, hash common.Hash) *types.PendingHeader {
 	key := pendingHeaderKey(hash)
 	data, _ := db.Get(key)
 	if len(data) == 0 {
 		return nil
```

Comment:
Perhaps add an error here as this will cause a nil pointer exception in the caller

Reply:
Returning nil here and the ReadPendingHeader user will make sure to check if the object is nil

Reply:
Logging an error now